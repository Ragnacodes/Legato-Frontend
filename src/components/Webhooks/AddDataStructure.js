import React, { useState } from 'react';
import {
  TextField,
  Button,
  Card,
  CardActions,
  CardContent,
  Typography,
} from '@material-ui/core';

const AddDataStructure = ({ setVisible }) => {
  const [dsInfo, setDsInfo] = useState({
    items: [
      { key: 1, tye: 'Array', label: 'item1' },
      { key: 2, type: 'Text', label: 'item2' },
      { key: 3, type: 'Text', label: 'item2' },
    ],
  });
  const handleDelete = (chipToDelete) => {
    setDsInfo((prev) => {
      return {
        ...prev,
        items: prev.items.filter((chip) => chip.key !== chipToDelete.key),
      };
    });
  };
  const handleChangeName = (e) => {
    setDsInfo((prev) => {
      return {
        ...prev,
        name: e.target.value,
      };
    });
  };
  return (
    <div className="ds-popover">
      <Typography className="ds-popover-title" variant="h5">
        Add Data Structure
      </Typography>
      <TextField
        className="edit-wh-field"
        onChange={handleChangeName}
        name="name"
        label="Data Structure Name"
        variant="outlined"
        size="small"
        defaultValue={dsInfo['name']}
        error={'name' in dsInfo && !dsInfo['name']}
        helperText={'name' in dsInfo ? !dsInfo['name'] && 'Required.' : ''}
      />
      <Typography variant="h6" component="h2">
        Items
      </Typography>
      <Card className="ds-items-card" variant="outlined">
        <CardContent className="ds-items-card-content">
          {dsInfo.items.map((item) => {
            return (
              <Card variant="outlined" className="ds-item">
                <CardContent>
                  {Object.keys(item).map((p) => (
                    <Typography variant="body2">{item[p]}</Typography>
                  ))}
                </CardContent>
              </Card>
              // <Chip
              //   size="small"
              //   className="ds-item"
              //   label={item.label}
              //   color="primary"
              //   onDelete={() => handleDelete(item)}
              // />
            );
          })}
        </CardContent>
        <CardActions>
          <Button variant="contained" color="primary" size="small">
            Add Item
          </Button>
        </CardActions>
      </Card>
      <div className="wh-ds-actions">
        <Button autoFocus onClick={() => setVisible(false)} color="primary">
          Cancel
        </Button>
        <Button
          onClick={() => {
            // handleSave(info);
            setVisible(false);
          }}
          variant="contained"
          color="primary"
        >
          Save
        </Button>
      </div>
    </div>
  );
};

export default AddDataStructure;
