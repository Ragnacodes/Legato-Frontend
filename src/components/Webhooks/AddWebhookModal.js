import React from 'react';
import {
  Typography,
  Dialog,
  DialogContent,
  DialogTitle,
} from '@material-ui/core';
import WebhookSettings from './WebhookSettings';

const AddWebhookModal = ({ visible, handleSave, setVisible }) => {
  const webhook = {
    name: "New Webhook",
    ip_restrictions: "",
    get_request_headers: false,
    get_request_http: false,
    json_passthrough: false,
  };
  return (
    <Dialog
      disableBackdropClick
      className="edit-wh-dialog"
      open={visible}
      onClose={() => setVisible(false)}
      aria-labelledby="edit-wh-dialog"
    >
      <DialogTitle disableTypography={true} className="edit-wh-dialog-title">
        <Typography variant="h5">Create a Webhook</Typography>
      </DialogTitle>
      <DialogContent>
        <WebhookSettings
          webhook={webhook}
          visible={visible}
          handleSave={handleSave}
          setVisible={setVisible}
        />
      </DialogContent>
    </Dialog>
  );
};

export default AddWebhookModal;
