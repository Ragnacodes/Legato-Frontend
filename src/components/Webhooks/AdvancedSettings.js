import React from "react";
import {
  Checkbox,
  FormControlLabel,
  Typography,
} from "@material-ui/core";
import InfoIcon from "@material-ui/icons/Info";
// import AddDataStructure from "./AddDataStructure";
const AdvancedSettings = ({ handleChange, info }) => {
  // const dsMenu = useRef(null);
  // const addDsPopper = useRef(null);
  // const [addDsPopperOpen, setAddDsPopperOpen] = useState(false);
  // const [dsMenuOpen, setDsMenuOpen] = useState(false);

  return (
    <div className="wh-advanced-settings">
      {/* <div className="data-structure-field">
        <FormControl
          className="edit-wh-field select-ds-field"
          size="small"
          variant="outlined"
        >
          <InputLabel id="demo-simple-select-outlined-label">
            Data structure
          </InputLabel>
          <Select
            labelId="demo-simple-select-outlined-label"
            id="demo-simple-select-outlined"
            value=""
            //   onChange={handleChange}
            label="Data structure"
          >
            <MenuItem value="">
              <em>None</em>
            </MenuItem>
            <MenuItem value={10}>Ten</MenuItem>
            <MenuItem value={20}>Twenty</MenuItem>
            <MenuItem value={30}>Thirty</MenuItem>
          </Select>
        </FormControl>
        <IconButton
          aria-label="edit-data-structures"
          fontSize="small"
          size="small"
          onClick={() => setDsMenuOpen(true)}
          // onMouseDown={(e) => e.preventDefault()}
        >
          <MoreVertIcon ref={dsMenu} />
        </IconButton>
        <Menu
          id="simple-menu"
          anchorEl={dsMenu.current}
          keepMounted
          open={dsMenuOpen}
          onClose={() => setDsMenuOpen(false)}
        >
          <MenuItem onClick={() => setDsMenuOpen(false)}>Edit</MenuItem>
          <MenuItem onClick={() => setDsMenuOpen(false)}>Clone</MenuItem>
        </Menu>
        <IconButton
          size="small"
          aria-label="add-data-structures"
          className="add-icon"
          onClick={() => setAddDsPopperOpen(true)}
          // onMouseDown={(e) => e.preventDefault()}
        >
          <AddIcon ref={addDsPopper} />
        </IconButton>
        <Popover
          disableBackdropClick
          open={addDsPopperOpen}
          anchorEl={addDsPopper.current}
          onClose={() => setAddDsPopperOpen(false)}
          anchorOrigin={{
            vertical: "top",
            horizontal: "right",
          }}
          transformOrigin={{
            vertical: "top",
            horizontal: "left",
          }}
        >
          <AddDataStructure setVisible={setAddDsPopperOpen} />
        </Popover>
      </div> 
      <Typography className="help-text" variant="body2">
        <InfoIcon fontSize="small" className="help-icon" />
        Data structure to be used for validation of incoming data. Leave empty
        if you want us to just pass on the received data without any validation.
      </Typography>
      */}
      <FormControlLabel
        control={
          <Checkbox
            name="get_request_headers"
            checked={info["get_request_headers"]}
            color="primary"
            onChange={(e) => handleChange(e.target.name, e.target.checked)}
          />
        }
        label="Get request headers"
      />
      <Typography className="help-text" variant="body2">
        <InfoIcon fontSize="small" className="help-icon" />
        The headers will be added to the request body.
      </Typography>
      <FormControlLabel
        control={
          <Checkbox
            name="get_request_http"
            checked={info["get_request_http"]}
            color="primary"
            onChange={(e) => handleChange(e.target.name, e.target.checked)}
          />
        }
        label="Get request HTTP method"
      />
      <Typography className="help-text" variant="body2">
        <InfoIcon fontSize="small" className="help-icon" />
        The HTTP method (verb) will be added to the request body.
      </Typography>
      <FormControlLabel
        control={
          <Checkbox
            name="json_passthrough"
            checked={info["json_passthrough"]}
            color="primary"
            onChange={(e) => handleChange(e.target.name, e.target.checked)}
          />
        }
        label="JSON pass-through"
      />
      <Typography className="help-text" variant="body2">
        <InfoIcon fontSize="small" className="help-icon" />
        If enabled, JSON payloads are returned as a string.
      </Typography>
    </div>
  );
};

export default AdvancedSettings;
