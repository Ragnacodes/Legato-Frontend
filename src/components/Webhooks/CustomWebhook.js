import React, { useState, useRef, useEffect } from 'react';
import { connect } from 'react-redux';
import * as actions from '../../actions/webhooks';
import {
  MenuItem,
  Typography,
  IconButton,
  TextField,
  Tooltip,
} from '@material-ui/core';
import Popover from '../Popover';
import { Info, Add, Edit } from '@material-ui/icons';
import { errorNotification, successNotification } from '../Layout/Notification';

import WebhookSettingsPopper from './WebhookSettingsPopper';

// export function HelperText(props) {
//   return <p className="url-helper-text"></p>;
// }
const CustomWebhook = ({
  anchor,
  handleSave,
  setAnchor,
  webhooks,
  getWebhooks,
  addWebhook,
  updateWebhook,
}) => {
  const [info, setInfo] = useState({
    webhook: '',
    max: '',
  });

  const [errors, setErrors] = useState({
    max: false,
    webhook: true,
  });

  useEffect(() => {
    getWebhooks()
      .then(() => {})
      .catch(() => {});
  }, [getWebhooks]);

  useEffect(() => {
    if (info['webhook']) {
      setInfo((prev) => ({
        ...prev,
        webhook: webhooks.find((w) => w.id === info['webhook'].id),
      }));
    }
  }, [webhooks]);

  const handleUpdateWebhook = (data) => {
    if (info['webhook']) {
      updateWebhook(info['webhook'].id, data)
        .then((res) => {
          successNotification(res.message);
        })
        .catch((err) => {
          errorNotification(err.message);
        });
    }
  };

  const addNewWebhook = (data) => {
    addWebhook(data)
      .then((res) => {
        successNotification(res.message);
      })
      .catch((err) => {
        errorNotification(err.message);
      });
  };

  useEffect(() => {
    console.log(info);
    if (!!info['webhook']) {
      setErrors((prev) => ({
        ...prev,
        webhook: false,
      }));
    }
  }, [info]);

  const handleChange = (e) => {
    setInfo((prev) => ({
      ...prev,
      [e.target.name]: e.target.value,
    }));
  };

  const CopyToClipboard = () => {
    navigator.clipboard.writeText(info['webhook'].url).then(
      function () {
        /* clipboard successfully set */
        setCopied(true);
      },
      function () {
        /* clipboard write failed */
      }
    );
  };

  const [editWhPopper, setEditWhPopper] = useState(null);
  const [addWhPopper, setAddWhPopper] = useState(null);
  const [copied, setCopied] = useState(false);

  const HelperTextCompoennt = React.forwardRef((props, ref) => {
    return (
      <Tooltip
        onClose={() => setCopied(false)}
        title={copied ? 'Copied!' : 'Copy to Clipboard'}
        placement="bottom"
        arrow
        classes={{
          popper: 'custom-helper-tooltip',
        }}
      >
        <span onClick={() => CopyToClipboard()} ref={ref} {...props} />
      </Tooltip>
    );
  });
  const FormHelperTextProps = {
    component: HelperTextCompoennt,
    classes: {
      root: 'url-helper-text MuiFormHelperText-root',
    },
  };
  return (
    <Popover
      className="custom-webhook"
      title="Custom Webhook"
      anchor={anchor}
      disabledSave={errors['webhook']}
      setAnchor={setAnchor}
      handleSave={() => {}}
      handleCancel={() => {}}
    >
      <div className="connection-field">
        <TextField
          disabled={Boolean(editWhPopper)}
          name="webhook"
          className="text-field"
          size="small"
          select
          label="Webhook"
          value={info['webhook']}
          onChange={handleChange}
          helperText={
            info['webhook'] ? info['webhook'].url : 'Please select a webhook.'
          }
          FormHelperTextProps={FormHelperTextProps}
          variant="outlined"
        >
          {webhooks.map((wh) => (
            <MenuItem key={wh.id} value={wh}>
              {wh.name}
            </MenuItem>
          ))}
        </TextField>

        <IconButton
          size="small"
          className="edit-icon"
          disabled={!info['webhook']}
          onClick={(e) => setEditWhPopper(e.currentTarget)}
        >
          <Edit />
        </IconButton>
        <WebhookSettingsPopper
          title={info['webhook'].name}
          webhook={info['webhook']}
          anchor={editWhPopper}
          setAnchor={setEditWhPopper}
          handleSave={handleUpdateWebhook}
        />
        <IconButton
          size="small"
          className="add-icon"
          onClick={(e) => setAddWhPopper(e.currentTarget)}
        >
          <Add />
        </IconButton>
        <WebhookSettingsPopper
          title="Add Webhook"
          webhook={{
            name: 'New Webhook',
          }}
          anchor={addWhPopper}
          setAnchor={setAddWhPopper}
          handleSave={addNewWebhook}
        />
      </div>

      <TextField
        name="max"
        className="text-field"
        label="Maximum number of results"
        type="number"
        variant="outlined"
        size="small"
        onChange={handleChange}
      />
    </Popover>
  );
};
const mapStateToProps = (state) => ({
  webhooks: state.webhooks.webhooks,
});

const mapDispatchToProps = (dispatch) => ({
  getWebhooks: () => dispatch(actions.startSetWebhooks()),
  addWebhook: (data) => dispatch(actions.startAddWebhook(data)),
  updateWebhook: (id, data) => dispatch(actions.startUpdateWebhook(id, data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(CustomWebhook);
