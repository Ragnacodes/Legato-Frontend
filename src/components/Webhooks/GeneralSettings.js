import React from "react";
import { TextField, Typography } from "@material-ui/core";

import InfoIcon from "@material-ui/icons/Info";

const GeneralSettings = ({ handleChange, info }) => {
  return (
    <div className="wh-general-settings">
      <TextField
        className="edit-wh-field"
        onChange={handleChange}
        name="name"
        label="Webhook Name"
        variant="outlined"
        size="small"
        defaultValue={info["name"]}
        error={!info["name"]}
        helperText={!info["name"] && "Required."}
      />
      <TextField
        className="edit-wh-field"
        onChange={handleChange}
        name="ip_restrictions"
        label="IP restrictions"
        variant="outlined"
        size="small"
      />
      <Typography className="help-text" variant="body2">
        <InfoIcon fontSize="small" className="help-icon" />A whitelist of IP
        addresses delimited by comma. You can use CIDR notation to whitelist a
        subnet. Leave empty if you don't want to check the IP address.
      </Typography>
    </div>
  );
};

export default GeneralSettings;
