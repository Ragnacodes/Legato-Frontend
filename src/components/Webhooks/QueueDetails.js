import React, { useEffect } from 'react';
import {
  Paper,
  Typography,
  Table,
  TableBody,
  TableCell,
  TableRow
} from '@material-ui/core';

const QueueDetails = ({ data }) => {
  //   const dataNames = ;
  useEffect(() => {
    console.log(data);
  }, [data]);

  return (
    <div className="wh-queue-details">
      <Typography variant="h6">
        <b>Data</b>
      </Typography>
      <div className="details">
        {Object.keys(data).length ? (
          <Paper variant="outlined">
            <Table size="small" aria-label="a dense table">
              <TableBody>
                {Object.keys(data).map((k) => (
                  <TableRow key={k}>
                    <TableCell
                      className="data-key"
                      align="right"
                      component="th"
                      scope="row"
                    >
                      {k}
                    </TableCell>
                    <TableCell align="left">{data[k]}</TableCell>
                  </TableRow>
                ))}
              </TableBody>
            </Table>
          </Paper>
        ) : (
          <Typography variant="caption">(empty)</Typography>
        )}
      </div>
    </div>
  );
};

export default QueueDetails;
