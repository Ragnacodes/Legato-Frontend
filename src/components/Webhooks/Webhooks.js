import React, { useEffect, useState } from "react";
import { connect } from "react-redux";
import { Container, Divider, Button } from "@material-ui/core";
import List from "@material-ui/core/List";
import Webhook from "./Webhook";
import Axios from "../../utils/axiosConfig";
import { setWebhooks, addWebhook } from "../../actions/webhooks";
import Appbar from "../Layout/Appbar";
import AddWebhookModal from "./AddWebhookModal";
import { errorNotification, successNotification } from "../Layout/Notification";
import PageTitle from "../Layout/PageTitle";

const Webhooks = ({ webhooks, username, setWebhooks, addWebhook }) => {
  const [addModalVisible, setAddModalVisible] = useState(false);
  useEffect(() => {
    Axios.get(`/users/${username}/services/webhooks`)
      .then((response) => {
        setWebhooks(response.data.webhooks);
      })
      .catch((error) => {
        console.log(error);
      });
  }, [setWebhooks, username]);

  const addNewWebhook = (data) => {
    Axios.post(`/users/${username}/services/webhooks`, {
      name: data.name,
    })
      .then((response) => {
        // setWebhooks(response.data);
        console.log(response.data);
        let str = response.data.message;
        successNotification(str.charAt(0).toUpperCase() + str.slice(1) + ".");
        addWebhook(response.data.webhook);
      })
      .catch((error) => {
        console.log(error);
        if (error.response) {
          let str = error.response.data.message;
          errorNotification(
            "Unable to update: " +
              str.charAt(0).toUpperCase() +
              str.slice(1) +
              "."
          );
        } else {
          errorNotification("Unable to update: " + error.message);
        }
      });
  };

  return (
    <>
      <Appbar
        rightChildren={
          <Button
            variant="contained"
            color="secondary"
            onClick={() => setAddModalVisible(true)}
          >
            Add Webhook
          </Button>
        }
        leftChildren={<PageTitle title="Webhooks" />}
      />
      <main className="main">
        <div className="app-bar-spacer" />
        <div className="content-container">
          <AddWebhookModal
            visible={addModalVisible}
            setVisible={setAddModalVisible}
            handleSave={addNewWebhook}
          />
          <Container maxWidth="lg" className="webhooks-list">
            <List>
              {webhooks.map((w) => {
                return (
                  <div key={w.id}>
                    <Webhook webhook={w} />
                    <Divider />
                  </div>
                );
              })}
            </List>
          </Container>
        </div>
      </main>
    </>
  );
};

const mapStateToProps = (state) => ({
  webhooks: state.webhooks.webhooks,
  username: state.auth.username,
});

const mapDispatchToProps = (dispatch) => ({
  setWebhooks: (data) => dispatch(setWebhooks(data)),
  addWebhook: (data) => dispatch(addWebhook(data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Webhooks);
