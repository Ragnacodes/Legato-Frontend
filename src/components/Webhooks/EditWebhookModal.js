import React from "react";
import { connect } from "react-redux";
import {
  Typography,
  Dialog,
  DialogContent,
  DialogContentText,
  DialogTitle,
  DialogActions,
  TextField,
  Button,
  Tab,
  Tabs,
} from "@material-ui/core";
import HelpIcon from "@material-ui/icons/Help";
import InfoIcon from "@material-ui/icons/Info";
import CloseRoundedIcon from "@material-ui/icons/CloseRounded";
import AdvancedSettings from "./AdvancedSettings";
export function TabPanel(props) {
  const { children, value, index, className, ...other } = props;

  return (
    <div
      className={"edit-webhook-tab " + className}
      role="tabpanel"
      hidden={value !== index}
      id={`simple-tabpanel-${index}`}
      aria-labelledby={`simple-tab-${index}`}
      {...other}
    >
      {value === index && children}
    </div>
  );
}

const Webhooks = ({ webhook, visible, setVisible }) => {
  const [value, setValue] = React.useState(0);
  const handleClose = () => {};
  const handleSave = () => {};
  const handleChange = (event, newValue) => {
    setValue(newValue);
  };
  return (
    <Dialog
      disableBackdropClick
      className="edit-webhook-dialog"
      open={visible}
      onClose={() => setVisible(false)}
      aria-labelledby="edit-webhook-dialog-title"
    >
      <DialogTitle disableTypography={true} id="edit-webhook-dialog-title">
        <Typography variant="h5">{webhook.name}</Typography>
      </DialogTitle>
      <DialogContent>
        <CloseRoundedIcon
          className="close-icon"
          onClick={() => setVisible(false)}
        />
        <Tabs
          value={value}
          onChange={handleChange}
          indicatorColor="primary"
          textColor="primary"
          centered
        >
          <Tab label="Settings" />
          <Tab label="Advanced Settings" />
        </Tabs>
        <TabPanel className="settings" value={value} index={0}>
          <TextField
            className="edit-field"
            //   onChange={onChange}
            name="username"
            label="Webhook Name"
            variant="outlined"
            size="small"
            defaultValue={webhook.name}
            //   error={!!errors["username"]}
            //   helperText={errors["username"]}
          />
          <TextField
            className="edit-field"
            //   onChange={onChange}
            name="ip-restriction"
            label="IP restrictions"
            variant="outlined"
            size="small"
            //   error={!!errors["username"]}
            //   helperText={errors["username"]}
          />
          <Typography className="help-text" variant="body2">
            <InfoIcon fontSize="small" className="help-icon" />A whitelist of IP
            addresses delimited by comma. You can use CIDR notation to whitelist
            a subnet. Leave empty if you don't want to check the IP address.
          </Typography>
        </TabPanel>
        <AdvancedSettings value={value} />
      </DialogContent>
      <DialogActions>
        <Button autoFocus onClick={handleClose} color="primary">
          Cancel
        </Button>
        <Button onClick={handleSave} variant="contained" color="primary">
          Save
        </Button>
      </DialogActions>
    </Dialog>
  );
};

const mapStateToProps = (state) => ({
  //   webhooks: state.webhooks.webhooks,
});

const mapDispatchToProps = (dispatch) => ({});

export default connect(mapStateToProps, mapDispatchToProps)(Webhooks);
