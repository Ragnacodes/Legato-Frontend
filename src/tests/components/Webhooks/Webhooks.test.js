import React from 'react';
import { shallow } from 'enzyme';
import { Webhooks } from '../../../components/Webhooks/Webhooks';
let wrapper, props, webhooks, getWebhooks, addWebhook;
beforeEach(() => {
  webhooks = [];
  getWebhooks = jest.fn();
  addWebhook = jest.fn();
  props = {
    webhooks,
    getWebhooks,
    addWebhook,
  };
  wrapper = shallow(<Webhooks {...props} />);
});

test('should render Webhooks Page correctly with no webhook', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render Webhooks Page correctly with onr webhook', () => {
  webhooks = [
    {
      name: 'name',
      id: 1,
      url: 'url',
      isEnable: true,
    },
  ];
  wrapper = shallow(<Webhooks {...props} webhooks={webhooks} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Webhooks Page correctly with more than one webhook', () => {
  webhooks = [
    {
      name: 'name',
      id: 1,
      url: 'url',
      isEnable: true,
    },
    {
      name: 'name two',
      id: 2,
      url: 'url/22',
      isEnable: false,
    },
  ];
  wrapper = shallow(<Webhooks {...props} webhooks={webhooks} />);
  expect(wrapper).toMatchSnapshot();
});
