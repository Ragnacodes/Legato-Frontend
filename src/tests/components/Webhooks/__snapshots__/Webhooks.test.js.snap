// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Webhooks Page correctly with more than one webhook 1`] = `<ListLoader />`;

exports[`should render Webhooks Page correctly with no webhook 1`] = `<ListLoader />`;

exports[`should render Webhooks Page correctly with onr webhook 1`] = `<ListLoader />`;
